src/useDrilldown/useDrilldown.ts

```diff
@@ -0,0 +1,61 @@
+import { useMemo, useState, useEffect, Dispatch } from 'react'
+import {
+  drilldownFn,
+  replaceFn,
+  getHiddenSiblingsFn,
+  getVisibleSiblingsFn,
+  removeFn,
+  disableEnumerables,
+  separateVisibleAndHiddenChildren,
+  Children
+} from './helpers'
+import { TreeItem, ParentKeyAccessor, makeParentChildTree, CHILDREN } from '../utils/makeParentChildTree'
+import { ListItem } from '../utils'
+
+export type DrilldownItem<T> = T & TreeItem<T> & Children<T> & {
+  drilldown: () => void
+  replace: (newSibling: DrilldownItem<T>) => void
+  getHiddenSiblings: () => TreeItem<T>[]
+  getVisibleSiblings: () => TreeItem<T>[]
+  remove: () => void
+}
+
+export function getAndEnrichVisibleItems<T> (list: TreeItem<T>[], visible: string[], setVisible: Dispatch<string[]>): DrilldownItem<T>[] {
+
+  return visible.map((key: string): DrilldownItem<T> => {
+    const item = list.find(item => item.key === key)!
```

Comment:
Why not do filter on list instead? Then you only iterate over the list once -> now depending on which are visible you might make a lot of unnecessary iterations